source/validate_id.cpp

```diff
@@ -417,7 +417,7 @@ bool idUsage::isValid<SpvOpTypeArray>(const spv_instruction_t* inst,
   } else if (5 == constInst->words.size()) {
     uint64_t value =
         constInst->words[3] | ((uint64_t)constInst->words[4]) << 32;
-    bool signedness = constResultType->second.inst->words[3];
+    bool signedness = constResultType->second.inst->words[3] ? true : false;
```

Comment:
If you wish to avoid implicit casts to bool, the correct fix is, e.g,
```
bool foo = words[3] > 0;
```
or, e.g,
```
failed = (value & overflow_mask) != 0;
```